blog/2024-02-01-py-client-v4-release/index.mdx

```diff
@@ -0,0 +1,436 @@
+            language="py"
+        />
+    </TabItem>
+</Tabs>
+
+### Batch operations
+
+The new client provides a set of methods to configure batching. These methods help you to control the batching behavior - including a handy `rate_limit()` method to avoid going past any third-party API rate limits.
+
+| Method | Description | When to use |
+| :-- | :-- | :-- |
+| `dynamic` | The batch size is dynamically calculated by Weaviate. | Recommended starting point. |
+| `fixed_size` | The batch size is fixed to a size specified by a user. | When you want to control the batch size. |
+| `rate_limit` | The number of objects sent to Weaviate is rate limited (specified as n_objects per minute). | When you want to avoid hitting third-party vectorization API rate limits. |
+
+<Tabs groupId="languages">
+    <TabItem value="wcs" label="Dynamic">
+        <FilteredTextBlock
+            text={PythonCode}
+            startMarker="# START BatchDynamic"
+            endMarker="# END BatchDynamic"
+            language="py"
+        />
+    </TabItem>
+    <TabItem value="local" label="Fixed size">
+        <FilteredTextBlock
+            text={PythonCode}
+            startMarker="# START BatchFixedSize"
+            endMarker="# END BatchFixedSize"
+            language="py"
+        />
+    </TabItem>
+    <TabItem value="embedded" label="Rate limit">
+        <FilteredTextBlock
+            text={PythonCode}
+            startMarker="# START BatchRateLimit"
+            endMarker="# END BatchRateLimit"
+            language="py"
+        />
+    </TabItem>
+</Tabs>
+
+#### Error handling in batch operations
+
+The `v4` client introduces additional error handling operations. You can now monitor the number of errors in an ongoing batching operation, or simply review the errors after the operation has completed.
+
+<Tabs groupId="languages">
+    <TabItem value="progress" label="During batching">
+        <FilteredTextBlock
+            text={PythonCode}
+            startMarker="# START BatchErrorMonitor"
+            endMarker="# END BatchErrorMonitor"
+            language="py"
+        />
+    </TabItem>
+    <TabItem value="end" label="After completion">
+        <FilteredTextBlock
+            text={PythonCode}
+            startMarker="# START BatchSimpleErrorHandling"
+            endMarker="# END BatchSimpleErrorHandling"
+            language="py"
+        />
+    </TabItem>
+</Tabs>
+
+### Insertion methods
+
+There's also single insertion methods, of course. And to that, we've added a `insert_many` method which is a convenience method for inserting a list of objects.
+
+<Tabs groupId="languages">
+    <TabItem value="batching" label="Batch">
+        <FilteredTextBlock
+            text={PythonCode}
+            startMarker="# START BatchBasic"
+            endMarker="# END BatchBasic"
+            language="py"
+        />
+        <i>Generally recommended for any multi-object operations.</i>
```

Comment:
why is the client example commented out?

Reply:
Oh yeah. I initially meant to highlight that they're alternatives; but you are right they can both exist in the same code.